operator/controllers/keptnevaluation/controller.go

```diff
@@ -118,11 +115,32 @@ func (r *KeptnEvaluationReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		if err != nil {
 			if errors.IsNotFound(err) {
 				r.Log.Info(err.Error() + ", ignoring error since object must be deleted")
+				span.SetStatus(codes.Error, err.Error())
 				return ctrl.Result{Requeue: true, RequeueAfter: 30 * time.Second}, nil
 			}
 			r.Log.Error(err, "Failed to retrieve a resource")
+			span.SetStatus(codes.Error, err.Error())
 			return ctrl.Result{}, nil
 		}
+		// load the provider
+		var provider KeptnSLIProvider
```

Comment:
during the refactoring, I would suggest to also extract the functionality of creating the EvaluationProvider into a separate component, i.e. a Factory. This will make it easier to unit test later